Fix incorrectly displayed UI elements on Windows

Chocobo1 opened the pull request:
Fix bugs in #2963.

### The pictures are huge, click to open it.

## Add new torrent window
Old | New
------------- | -------------
[add_torrent_old_1](https://cloud.githubusercontent.com/assets/9395168/11587237/fc0daa08-9ab2-11e5-965e-1096196bffd4.png) | [add_torrent_new_1](https://cloud.githubusercontent.com/assets/9395168/11587144/5f9222da-9ab2-11e5-9c21-06e57711a842.png)


## options window
Old | New
------------- | -------------
[adv_opt_old_1](https://cloud.githubusercontent.com/assets/9395168/11587236/fc0b4786-9ab2-11e5-8f71-774aec904688.png) | [adv_opt_new_1](https://cloud.githubusercontent.com/assets/9395168/11587145/5f9812a8-9ab2-11e5-9aef-ebc17ddca16c.png)


## about dialog
Old | New
------------- | -------------
[opt_old_1](https://cloud.githubusercontent.com/assets/9395168/11587278/343d1652-9ab3-11e5-8570-9f02cad1928e.png) | [opt_new_1](https://cloud.githubusercontent.com/assets/9395168/12222553/d7de6cca-b7fa-11e5-835f-5a8734ef5a5b.png)
[opt_old_2](https://cloud.githubusercontent.com/assets/9395168/11587279/344039cc-9ab3-11e5-9e2e-eb18b6b19bdb.png) | [opt_new_2](https://cloud.githubusercontent.com/assets/9395168/12000434/5f42f56a-ab11-11e5-8ad6-1ec5bdf30f74.png)
[opt_old_3](https://cloud.githubusercontent.com/assets/9395168/11587281/344384ec-9ab3-11e5-83fb-de7680275e83.png) | [opt_new_3](https://cloud.githubusercontent.com/assets/9395168/12000430/5f3ca0d4-ab11-11e5-8157-f06c90b38601.png)
[opt_old_4](https://cloud.githubusercontent.com/assets/9395168/11587280/34436bba-9ab3-11e5-9e69-7f11de40cd15.png) | [opt_new_4](https://cloud.githubusercontent.com/assets/9395168/12000431/5f3e6d56-ab11-11e5-8732-c6cbb266d01d.png)
[opt_old_5](https://cloud.githubusercontent.com/assets/9395168/11587282/34442d70-9ab3-11e5-8a69-1793c2fe633b.png) | [opt_new_5](https://cloud.githubusercontent.com/assets/9395168/12000432/5f3f75fc-ab11-11e5-9191-7b3badb43fb8.png)
[opt_old_6](https://cloud.githubusercontent.com/assets/9395168/11587283/3445a100-9ab3-11e5-9c82-688bb537b159.png) | [opt_new_6](https://cloud.githubusercontent.com/assets/9395168/12000433/5f3ffec8-ab11-11e5-9ca3-46e397cb22ee.png)

Chocobo1 commented:
This is ready for review. If changes are not preferred, can revert.

ngosang commented:
I agree with all changes (I tested it but I didn't see the code).
@Chocobo1 Several optional improvements.
![](http://i.imgur.com/uUxyMjv.png)
1) For me the title size is too large.
2) The icon size is a bit small.
3) Emails should be mailto: links (or at least you should be able to copy them)
![](http://i.imgur.com/ygIwQDR.png)
1) Remove the dots.
2) You should be able to copy the libraries to the clipboard and text should be left aligned (as Authors tab).

Chocobo1 commented:
>I agree with all changes (I tested it but I didn't see the code).
@Chocobo1 Several optional improvements.

I've done all your suggestions except "copy the libraries to the clipboard", I'll consider to add it later.
First post images updated.

ngosang commented:
rebase?

Chocobo1 commented:
>rebase?

Done.

ngosang commented:
@Chocobo1 Good work, looks perfect.
Just one detail. Could you align the links here:
![](http://storage3.static.itmages.com/i/16/0110/h_1452437587_7482128_26e4acebcd.png)
as you did here?
![](https://cloud.githubusercontent.com/assets/9395168/12000434/5f42f56a-ab11-11e5-8ad6-1ec5bdf30f74.png)

Chocobo1 commented:
>Just one detail. Could you align the links here:

Done. I put them under \<table> tag.

ngosang commented:
@sledgehammer999 I think this is ready.

LordNyriox commented:
@Chocobo1  This PR reports conflicts.  Do you think you could rebase? 

Thanks, 
~ Nyriox

LordNyriox commented:
@sledgehammer999 Any chance of merging this?  This fixes some UI styling that's been bugging me for a long time, and @Chocobo1 hasn't had much time to maintain his PRs lately.  

Thanks, 
~ Nyriox

ngosang commented:
@Chocobo1 rebase

Chocobo1 commented:
>@Chocobo1 rebase

Done.

LordNyriox commented:
@sledgehammer999 Might this PR make it into the next release?  Like [https://github.com/qbittorrent/qBittorrent/pull/4122], this PR mostly just moves UI elements around and fixes various scaling issues.  

Thanks, 
~ Nyriox

sledgehammer999 commented:
Add torrent dialog: "Always auto confirm" reads funny. I think the old string was better.
Options: OK
About: OK but the 'Libraries' page doesn't look good. I think the old way was better.

Of course, I judge this from your screenshots. Sometime I'll test myself too.

Offtopic: Your font renders greek letters very weird (`Στέφανος Αντάρης` in Thanks page)

Chocobo1 commented:
>Add torrent dialog: "Always auto confirm" reads funny. I think the old string was better.

will revert.

>About: OK but the 'Libraries' page doesn't look good. I think the old way was better.

you want to revert all or just move the (lib_name: version) table to the horizontal center?

>Offtopic: Your font renders greek letters very weird (Στέφανος Αντάρης in Thanks page)

I always wonder why Qt won't use my system default font for everything... (the system default font is the one used for window title). Probably a Qt bug.

LordNyriox commented:
@Chocobo1 On the Libraries about page:  I think the problem is more along the lines of spacing as relates to the 'libtorrent-rasterbar' label.  In the old version, the text looked balanced, but not so much with your (much longer) edited name.  

Why don't you try reverting '**libtorrent-rasterbar**' back to '**Libtorrent**' and post an updated screenshot?  I really think that's the only problem.  

Thanks, 
~ Nyriox

LordNyriox commented on src/gui/about.ui:
```diff
@@ -394,125 +301,106 @@
+          <item row="1" column="0">
+           <widget class="QLabel" name="label_9">
+            <property name="text">
+             <string notr="true">libtorrent-rasterbar:</string>
```
@Chocobo1 When in a table with the much shorter labels '**Qt**' and '**Boost**', '**libtorrent-rasterbar**' looks much too long.  Try renaming it back to '**Libtorrent**', then ask @sledgehammer999 what he thinks of the changes then.

sledgehammer999 commented:
>you want to revert all or just move the (lib_name: version) table to the horizontal center?

Can you provide a screenshot of how it looks with just moving the table to horizontal center?
PS: make `libtorrent-rasterbar` read `Libtorrent-rasterbar`(capital L ) too.

>I always wonder why Qt won't use my system default font for everything... (the system default font is used for window title in screenshot). Probably a Qt bug.

Do you use windows?

Chocobo1 commented:
![pic](https://cloud.githubusercontent.com/assets/9395168/12864767/a4984a2a-ccd1-11e5-8931-1b1fb2defb4c.png)

>Do you use windows?
Of course.

LordNyriox commented:
@Chocobo1 I'd recommend increasing the spacing between the table columns.  Right now `Libtorrent-rasterbar` is smashed right up against its version number, making that line virtually unreadable.  Could you attempt that and post another screenshot? 

Thanks, 
~ Nyriox

Chocobo1 commented:
>making that line virtually unreadable

What line? There is no line in this table.

LordNyriox commented:
@Chocobo1 I meant row.  Specifically, the row of text containing `Libtorrent-rasterbar`.  I find it difficult to read with the tight spacing.

Chocobo1 commented:
> I find it difficult to read with the tight spacing. 

Probably due to the font used. If your system don't use `Times New Roman` font for every labels then you won't be affected by this issue. I mean it will look ok on your system, so no changes here.

LordNyriox commented:
@Chocobo1 That very well might be the case, but I doubt your situation is unique.  All I'm asking is for you to add a blank column to the middle of your table, 1-2 spaces wide.  Though truth be told I think about twice that would be better for visual balance with the window.  

Thanks, 
~ Nyriox

sledgehammer999 commented:
@Chocobo1 "Libraries" tab still looks weird. Revert to the old one and I can merge then.

sledgehammer999 commented:
Offtopic: Why I can't see @LordNyriox's recent comments, but receive them in the email notifications?

LordNyriox commented:
@sledgehammer999 OFF-TOPIC: Because I delete them after I feel they've made their necessary impact.  I don't want to clutter up the site, and I figure anyone who cares will have an e-mail copy anyway.  The only exceptions I make are cases where I truly feel like I've contributed meaningfully somehow.  

Thanks for your concern, however! 

~ Nyriox

Chocobo1 commented:
>@Chocobo1 "Libraries" tab still looks weird. Revert to the old one and I can merge then.

Done.

LordNyriox commented:
@sledgehammer999 The Libraries tab is now reverted, and no conflicts are reported.  I believe this PR is ready to merge. 

Thanks, 
~ Nyriox

sledgehammer999 commented:
@Chocobo1 I was almost going to merge this but I found a small regression. The "add new torrent dialog" can't be resized in width under a certain width and that width is too big. The current minimal width is pretty good IMO.
**This isn't required for merging:** Can you make the QCombobox for the labels to not always resize with the extra width of the dialog? It becomes silly after a certain big width.

Chocobo1 commented:
>I was almost going to merge this but I found a small regression. The "add new torrent dialog" can't be resized in width under a certain width and that width is too big. The current minimal width is pretty good IMO.

I found a different behavior for qcombobox on windows & linux. On linux the qcombobox can be freely resize to width 1, but on windows it's limited to content's width.
I changed `SizeAdjustPolicy` to `AdjustToMinimumContentsLength`, please try it out, not sure this solves the problem.

>This isn't required for merging: Can you make the QCombobox for the labels to not always resize with the extra width of the dialog? It becomes silly after a certain big width.

Done.


Besides these fixes, I added 1 new commit. It simplifies the resize actions and also minimize flickering when doing resize.

sledgehammer999 commented:
The resizing works great now and maybe your last commit fixes the problem with duplicate buttons(+magnets). However, I think the placement of the central widgets when the dialog is very wide becomes weird. There's a lot of space left at the right. Current behavior is to leave it at the middle. What do you say, is it weird? (pic follows)

![untitled](https://cloud.githubusercontent.com/assets/273315/13063647/ec2ca6d2-d452-11e5-8959-e840e1271608.jpg)

If not, then I suggest to at least add a bitmore padding between the columns of the central widhets. When the dialog becomes very wide they look very tight.

sledgehammer999 commented:
PS: I am talking about the widgets in the `Torrent Settings` container.

Chocobo1 commented:
>The resizing works great now and maybe your last commit fixes the problem with duplicate buttons(+magnets). 

I never encounter that issue. Hope it's fixed.

>There's a lot of space left at the right. Current behavior is to leave it at the middle. What do you say, is it weird? (pic follows)

Yeah, it's too close & feels unnatural. I did a fix up by adding a horizontal spacer:
![image](https://cloud.githubusercontent.com/assets/9395168/13066639/2e285248-d4a3-11e5-8183-2be70f5963b0.png)

sledgehammer999 commented:
Everything works fine now. Thank you very much.
